From: Scott Moser <smoser@ubuntu.com>
Subject: fix lxc_driver when running on linux 3.0
Bug: https://bugs.launchpad.net/ubuntu/+source/libvirt/+bug/802977
Last-updated: 2011-06-28
Forwarded: yes

linux 3.0 has no micro version number, and that is causing problems.

This should allow for
  major
  major.minor
  major.minor.micro

Anything not supplied defaults to 0

--- a/src/util/util.c
+++ b/src/util/util.c
@@ -2241,17 +2241,17 @@ virParseNumber(const char **str)
 int
 virParseVersionString(const char *str, unsigned long *version)
 {
-    unsigned int major, minor, micro;
+    unsigned int major, minor=0, micro=0;
     char *tmp;
 
-    if (virStrToLong_ui(str, &tmp, 10, &major) < 0 || *tmp != '.')
+    if (virStrToLong_ui(str, &tmp, 10, &major) < 0)
         return -1;
 
-    if (virStrToLong_ui(tmp + 1, &tmp, 10, &minor) < 0 || *tmp != '.')
-        return -1;
+    if ((*tmp == '.') && virStrToLong_ui(tmp + 1, &tmp, 10, &minor) < 0)
+            return -1;
 
-    if (virStrToLong_ui(tmp + 1, &tmp, 10, &micro) < 0)
-        return -1;
+    if ((*tmp == '.') && virStrToLong_ui(tmp + 1, &tmp, 10, &micro) < 0)
+            return -1;
 
     *version = 1000000 * major + 1000 * minor + micro;
 
